events/wc_crisis_events.txt

```diff
@@ -0,0 +1,736 @@
+namespace = WCCRS
+
+### CRISIS: DRUST INCURSION EVENTS ###
+# Opening event: A sick spouse
+character_event = {
+	id = WCCRS.100
+
+	is_triggered_only = yes
+	hide_window = yes
+
+	only_playable = yes
+
+	trigger = {
+		trigger_if = {
+			limit = { drust_sick_spouse_trigger = yes }
+			any_spouse = { drust_worried_spouse_trigger = yes }
+		}
+		trigger_else = {
+			drust_worried_spouse_trigger = yes
+			any_spouse = { drust_sick_spouse_trigger = yes }
+		}
+	}
+
+	immediate = {
+		set_global_flag = heartsbane_coven_event_flag
+		set_global_flag = drustvar_crisis_ongoing_flag	# Blocks summon_gorak_tul
+		
+		# Picks who is sick and who isn't
+		if = {
+			limit = { drust_sick_spouse_trigger = yes }
+			save_event_target_as = target_sick_spouse
+			random_spouse = {
+				limit = { drust_worried_spouse_trigger = yes }
+				save_event_target_as = target_worried_spouse
+			}
+		}
+		else = {
+			save_event_target_as = target_worried_spouse
+			random_spouse = {
+				limit = { drust_sick_spouse_trigger = yes }
+				save_event_target_as = target_sick_spouse
+			}
+		}
+		
+		event_target:target_worried_spouse = { narrative_event = { id = WCCRS.110 } }
+	}
+}
+
+#Drust King, Gorak Tul contacts the grief-stricken spouse
+narrative_event = {
+	id = WCCRS.110
+	title = EVTTITLE_WCCRS_110
+	desc = EVTDESC_WCCRS_110
+	picture = GFX_evt_drustvar
+	border = GFX_event_narrative_frame_intrigue
+
+	is_triggered_only = yes
+
+	immediate = {
+		log = "WCCRS.110 fired for [Root.GetBestName]"
+		
+		# Moved to WCCRS.119
+	}
+
+	# Not even death will do us part
+	option = {
+		name = EVTOPTA_WCCRS_110
+		
+		event_target:target_worried_spouse = {
+			add_trait_undead_effect = yes			# This effect sets portrait
+			religion = throsic		# Need to set undead religion before necromantic classes can be assigned
+			if = {
+				limit = {
+					potential_necromancer_trigger = yes
+				}
+				# replace old class with necromancer
+			}
```

Comment:
Death knight effect should go first, necromancer next. Otherwise, if character is mage + warrior, it will get necromancer class only.
```suggestion
			if = {
				limit = {
					is_death_knight_class_trigger = no
					OR = {
						is_physical_class_trigger = yes
						is_combined_class_trigger = yes
					}
					potential_death_knight_trigger = yes
				}
				add_character_modifier = { name = training_self_death_knight duration = -1 }
				recount_experience_effect = yes
				add_class_trait_effect = yes
			}
```

Reply:
Actually it ended up just adding mage's experience to the DK class instead, so if you have level 5 dk + level 3 mage, it ends up becoming level 8 DK.

Reply:
You can't become a mage being a dk.

Reply:
alright, everything else seems fine, pushing